File: Python/PlenoInterview/Problem1/order.py
class order:

    def __init__(self, numPeople):
        self.orderItems = []
        self.numPeople = numPeople

    def getOrderTotal(self):
        total = 0
        for order in self.orderItems:
            total += order.getTotalPrice()
        return total

    def getTotalPerPerson(self):
        return self.getOrderTotal()/self.numPeople

    def getOrderItemByName(self, dishName):
        for order in self.orderItems:
            if (order.dishName == dishName):
                return order
        return None

    def addNewOrderItem(self, orderItem):
        self.orderItems.append(orderItem)

    def addExistingOrderItem(self, orderItem):
        oi = self.getOrderItemByName(orderItem.dishName)
        oi.amount += 1

    def getTotalPrice(self, orderItem):
        return orderItem.getTotalPrice()

    def printOrder(self):
        self.orderItems.sort(key=lambda oi : oi.getTotalPrice(), reverse=True)
        for order in self.orderItems:
            print(order)
        print("===================")
        print("Total: {:.2f}".format(self.getOrderTotal()))
        print("Num of people: {}".format(self.numPeople))
        print("Total per person: {:.2f}".format(self.getTotalPerPerson()))

File: Problems/Implementation/I_2_restaurant2/resources/Restaurant_Implementation/Order.java
import java.util.List;
import java.util.ArrayList;

class Order{
    //VARIABLES
    private List<OrderItem> orderItems;
    private int NumPeople;

    // CONSTRUCTOR
    public Order(int numPeople){
        this.orderItems = new ArrayList<OrderItem>();
        this.NumPeople = numPeople;
    }

    //region GETTER/SETTER
    public List<OrderItem> getOrderItems() {
        return orderItems;
    }

    public void setOrderItems(List<OrderItem> orderItems) {
        this.orderItems = orderItems;
    }

    public int getNumPeople() {
        return NumPeople;
    }

    public void setNumPeople(int numPeople) {
        this.NumPeople = numPeople;
    }
    //endregion

    public double getOrderTotal(){
        double total = 0;
        for (OrderItem oi : orderItems){
        total += oi.getTotalPrice();
        }
        return total;
    }

    public double getTotalPerPerson(){
        return getOrderTotal()/this.NumPeople;
    }

    public OrderItem getOrderItemByName(String dishName){
        for (OrderItem oi : this.orderItems){
            if (oi.getDishName().equals(dishName)) {
                return oi;
            }
        }
        return null;
    }

    public void addNewOrderItem(OrderItem oi){
        orderItems.add(oi);
    }

    public void addExistingOrderItem(OrderItem oiToAdd){
        OrderItem oi = getOrderItemByName(oiToAdd.getDishName());
        oi.setAmount(oi.getAmount()+1);
    }

    public void printOrder(){
        orderItems.sort((o1, o2) -> Double.compare(o2.getTotalPrice(), o1.getTotalPrice()));
        for (OrderItem oi : this.orderItems){
            System.out.println(oi);
        }
        System.out.println("=========================");
        System.out.printf("Total: $%.2f\n", getOrderTotal());
        System.out.printf("Num of people: %d\n", NumPeople);
        System.out.printf("Total per person: $%.2f\n", getTotalPerPerson());
    }
}